### PR TITLE
feat: add modal sample using already defined component

### PR DESCRIPTION
**Pull request by quentinderoubaix**

### CODE REVIEW

File: angular/lib/src/lib/modal/modal.component.ts [hunk 1]
```diff
@@ -213,25 +213,35 @@ export class ModalComponent implements OnChanges, AfterContentChecked {
...
 
-	@Input('auSlotStructure') slotStructure: SlotContent<ModalContext>;
+	// eslint-disable-next-line @agnos-ui/angular-check-props
+	@Input('auSlotStructure') slotStructure: SlotContent<ModalContext<Data>>;
 	@ContentChild(ModalStructureDirective, {static: false})
-	slotStructureFromContent: ModalStructureDirective | null;
+	slotStructureFromContent: ModalStructureDirective<Data> | null;
 
-	@Input('auSlotHeader') slotHeader: SlotContent<ModalContext>;
+	// eslint-disable-next-line @agnos-ui/angular-check-props
```
Comment (quentinderoubaix) [inline angular/lib/src/lib/modal/modal.component.ts:221]: @divdavem the angular-check-props plugin was failing and am not really sure why...

Comment (quentinderoubaix) [reply angular/lib/src/lib/modal/modal.component.ts]: Thx !

File: react/lib/modal/Modal.tsx [hunk 2]
```diff
@@ -1,70 +1,76 @@
...
-			</Portal>
-		</>
+				</div>
+			)}
+		</Portal>
 	);
-});
+}
+
+export const Modal = forwardRef(ModalInner) as <Data>(
```
Comment (divdavem) [inline react/lib/modal/Modal.tsx:74]: Why did you introduce the `ModalInner` function instead of keeping the concise inline syntax?
Why did you change the arrow function syntax (in all functions in this file) to use the more verbose function syntax?

File: react/lib/modal/Modal.tsx [hunk 3]
```diff
@@ -1,70 +1,76 @@
...
+	props: PropsWithChildren<Partial<ModalProps<Data>>> & {ref?: Ref<unknown> | undefined}
```
Comment (divdavem) [inline react/lib/modal/Modal.tsx:75]: `Ref<unknown>` is suspicious to me. Why don't you use the correct type `Ref<ModalWidget<Data>['api']>`?

**Review by divdavem — CHANGES_REQUESTED** [review]
Thank you @quentinderoubaix for your PR.
It is already very good but it would be even better if you could fix the few things I have commented.

File: react/lib/modal/Modal.tsx (hunk 2 above)
Comment (quentinderoubaix) [reply react/lib/modal/Modal.tsx:74]: For the ```ModalInner```, I followed a recommendation (https://fettblog.eu/typescript-react-generic-forward-refs/) as forwardRef and generic component don' t work well together.
I will put back the other functions to arrow syntax

File: react/lib/modal/Modal.tsx (hunk 3 above)
Comment (quentinderoubaix) [reply react/lib/modal/Modal.tsx]: Thx for the help, it is not correctly typed.

File: react/lib/modal/Modal.tsx (hunk 2 above)
Comment (quentinderoubaix) [reply react/lib/modal/Modal.tsx]: ModalInner removed too.

File: demo/scripts/includeSamples.plugin.ts [hunk 4]
```diff
@@ -54,7 +54,11 @@ export const includeSamples = (): Plugin => {
 						for (const dependency of dependencies) {
 							const dependencyParts = dependency.split('/');
 							if (dependencyParts[0] === '.') {
-								await addFile(framework, path.basename(dependency), path.join(directory, dependency));
+								let relativePath = dependency;
+								if (dependency.endsWith('.component') && framework === 'angular') {
```
[the remark this answers is not in the document]
Comment (quentinderoubaix) [reply demo/scripts/includeSamples.plugin.ts]: Condition remade to be framework independant.

**Comment by quentinderoubaix** [comment]
Have made the proposed changes, thx for the review @divdavem !

**Review by divdavem — APPROVED** [review]
Thank you! :+1:
